blocks/graph.py

```diff
@@ -205,8 +205,9 @@ def replace(self, replacements):
         for node in apply_nodes_sorted:
```

Comment:
Thinking of it, it's not, unless the user happens to try to replace an auxiliary variable.